sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -2523,6 +2532,7 @@ def __init__(
       use_at_least_once=False,
       with_auto_sharding=False,
       num_storage_api_streams=0,
+      cdc_writes_with_primary_key: List[str] = None,
```

Comment:
Non-blocking if you're trying to get this in soon, but I was thinking we could allow users to specify either a boolean or a callable for `use_cdc_writes`:
- If they provide a boolean, we pass their records directly to the Java IO
- If they provide a callable, we pass their rows to the callable then to the Java IO

Either way, we make it clear that the resulting rows should have the appropriate `cdc_info` fields. 

Could be a future improvement if the current use-case is sufficient for now.

Reply:
added the callable argument to the constructor. 

I enforced the types for the callable to make sure we already have a Row that will be used to extract the CDC info of it.

Reply:
`beam.Row` is shorthand. You can find the module at `apache_beam.pvalue.Row`

Anyway, I think we can push this callable option to another PR because I suspect it will take a good amount of work. I suggest we drop this so that we can merge this change soon and make it available in the upcoming release (which will be cut this Wednesday)